File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/AspectRatio.kt
package com.kinetixui.ui

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * KinetixAspectRatio — mirrors `packages/ui/src/components/aspect-ratio.tsx`,
 * itself a bare re-export of Radix's `AspectRatio.Root` with no styling of
 * its own. Compose has a direct built-in equivalent (`Modifier.aspectRatio`)
 * — this just gives it a `Kinetix`-prefixed, `content`-slot-shaped API
 * consistent with the rest of the package.
 */
@Composable
fun KinetixAspectRatio(
    ratio: Float,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Box(modifier = modifier.aspectRatio(ratio)) {
        content()
    }
}

File: packages/ui-swiftui/Sources/KinetixUI/AspectRatio.swift
//
// AspectRatio.swift — KinetixAspectRatio.
//
// Mirrors packages/ui/src/components/aspect-ratio.tsx, itself a bare
// re-export of Radix's `AspectRatio.Root` with no styling. SwiftUI has a
// direct built-in (`.aspectRatio`) — this just gives it a `Kinetix`-
// prefixed, content-slot API consistent with the rest of the package.
//

import SwiftUI

public struct KinetixAspectRatio<Content: View>: View {
    private let ratio: CGFloat
    private let content: Content

    public init(_ ratio: CGFloat, @ViewBuilder content: () -> Content) {
        self.ratio = ratio
        self.content = content()
    }

    public var body: some View {
        Color.clear
            .aspectRatio(ratio, contentMode: .fit)
            .overlay { content }
            .clipped()
    }
}
